File: from-0001-to-0100/0094-binary-tree-inorder-traversal/solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
#         res = []
#         curr = root
#         stack = []
        
#         while curr or stack:
#             while curr:
#                 stack.append(curr)
#                 curr = curr.left
                
#             curr = stack.pop()
#             res.append(curr.val)
#             curr = curr.right
            
#         return res

        result = []
        def inorder(node):
            nonlocal result
            if not node:
                return
            
            inorder(node.left)
            result.append(node.val)
            inorder(node.right)
            
        inorder(root)
        
        return result

File: from-0001-to-0100/0094-binary-tree-inorder-traversal/solution.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
 type stackTreeNode []*TreeNode

 func (s *stackTreeNode) push(node *TreeNode) {
	 *s = append(*s, node)
 }
 
 func(s *stackTreeNode) pop() *TreeNode {
	 t := (*s)[len(*s) - 1]
	 *s = (*s)[:len(*s) - 1]
	 return t
 }
 
 func (s stackTreeNode) isEmpty() bool {
	 return len(s) == 0
 }
 
 func inorderTraversal(root *TreeNode) []int {
	 if root == nil {
		 return nil
	 }
	 result := []int{}
	 curr := root
	 
	 s := stackTreeNode{}
 
	 for curr != nil || !s.isEmpty() {
		 for curr != nil {
			 s.push(curr)
			 curr = curr.Left
		 }
		 curr = s.pop()
		 result = append(result, curr.Val)
		 curr = curr.Right
	 }
	 return result
 }